app/src/main/java/com/cornellappdev/android/eatery/ui/screens/FavoritesScreen.kt

```diff
@@ -139,118 +139,157 @@ fun FavoritesScreen(
-                                    onFavoriteClick = {
-                                        if (!it) {
-                                            favoriteViewModel.removeFavorite(eatery.id)
-                                        }
-                                    }) {
-                                    onEateryClick(it)
-                                }
-                            }
-                        } else {
-                            items(favoritesScreenViewState.favoriteCards) { itemFavoritesCardViewState ->
-                                ItemFavoritesCard(
-                                    itemFavoritesCardViewState
-                                )
+                    if (toggle) {
+                        item {
+                            AnimatedVisibility(favoriteEateries.isEmpty()) {
+                                EateriesEmptyState(message = "You currently have no favorite eateries!")
                             }
                         }
 
+                        items(
+                            items = favoriteEateries,
+                            key = { eatery ->
+                                eatery.id!!
+                            }) { eatery ->
+
+                            EateryCard(
+                                eatery = eatery,
+                                isFavorite = true,
+                                modifier = Modifier.animateItemPlacement(),
+                                onFavoriteClick = {
+                                    if (!it) {
+                                        favoriteViewModel.removeFavorite(eatery.id)
+                                    }
+                                }) {
+                                onEateryClick(it)
+                            }
+                        }
+                    } else {
                         item {
-                            Spacer(Modifier.height(20.dp))
+                            AnimatedVisibility(favoritesScreenViewState.favoriteCards.isEmpty()) {
+                                EateriesEmptyState("You currently have no favorite menu items!")
+                            }
                         }
+                        items(favoritesScreenViewState.favoriteCards) { itemFavoritesCardViewState ->
+                            ItemFavoritesCard(
+                                itemFavoritesCardViewState,
+                                modifier = Modifier.animateItemPlacement(),
+                                onFavoriteClick = {
+                                    favoriteViewModel.removeFavoriteMenuItem(
+                                        itemFavoritesCardViewState.itemName
+                                    )
+                                }
+                            )
+                        }
+                    }
+
+                    item {
+                        Spacer(Modifier.height(20.dp))
                     }
```

Comment:
Same thing for this! And then that also allows us to perhaps change the if-else to a pretty easily readable `AnimatedContent` (if that looks good, idk)